### ISSUE TITLE
[Qt6, PyQt6 6.6.1] Breaks keyboard shortcuts layout in Preferences

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#6601]
### 🐛 Bug Report

With PyQt6 6.6.1 the keyboard shortcuts layout is broken with the action names prematurely truncated.

6.6.1
![image](https://github.com/napari/napari/assets/76622105/eaf95d73-7928-4c3a-830c-c254a4056841)

6.6.0 (and 6.5.3, 6.5.0)
![image](https://github.com/napari/napari/assets/76622105/97eac1cf-6593-448b-b46a-fc893541108e)

(note when testing old versions you need to force install the matching version of `PyQt6-Qt6`)

### 💡 Steps to Reproduce

Install latest PyQt6 (6.6.1) with matching PyQt6-Qt6 package.
Open the keyboard shortcuts Preference panel.

### 💡 Expected Behavior

The full action names should be shown.

### 🌎 Environment

napari: 0.5.0a2.dev483+g65ffc895
Platform: macOS-14.2.1-arm64-arm-64bit
System: MacOS 14.2.1
Python: 3.11.7 | packaged by conda-forge | (main, Dec 23 2023, 14:38:07) [Clang 16.0.6 ]
Qt: 6.6.1
PyQt6: 
NumPy: 1.26.3
SciPy: 1.11.4
Dask: 2024.1.0
VisPy: 0.14.1
magicgui: 0.8.1
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.3

OpenGL:
- GL version: 2.1 Metal - 88
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 2056x1329, scale 2.0
- screen 2: resolution 1920x1200, scale 1.0

Settings path:
- /Users/sobolp/Library/Application Support/napari/napari-dev-pyqt6_44a57b2eefd248abfe19ee6d23ba0bf2f2a31d6d/settings.yaml

### 💡 Additional Context

This is breaking behavior added by https://github.com/napari/napari/pull/5679

**Comment by psobolewskiPhD** [comment]
@dalthviz Do you have any ideas where to start to try to debug this?

**Comment by dalthviz** [comment]
Hi there, maybe something changed regarding word wrapping or maybe the way to make cells use all the space available :thinking: Will give this a check and let you know if I find something :+1:

**Comment by dalthviz** [comment]
@psobolewskiPhD I was able to reproduce this on Windows and the only way I have found (at least for the moment) to properly setup an initial column and row size was setting a maximum section size for the table horizontal header while also using the [`QHeaderView.ResizeMode.ResizeToContents`](https://doc.qt.io/qt-6.5/qheaderview.html#ResizeMode-enum) mode for the table horizontal and vertical headers. So changing the code at:

https://github.com/napari/napari/blob/0be7f18fb67d4f68876a0e5ca04c619dd343f456/napari/_qt/widgets/qt_keyboard_settings.py#L233-L238

For something like:

```python
            # Column and rows size set up.
            self._table.horizontalHeader().setMaximumSectionSize(370)
            self._table.horizontalHeader().setSectionResizeMode(
                QHeaderView.ResizeMode.ResizeToContents
            )
            self._table.verticalHeader().setSectionResizeMode(
                QHeaderView.ResizeMode.ResizeToContents
            )
```

Makes things have a proper size even when using PyQt6 6.6.1, PyQt6 6.6.0 or PyQt5 5.15.9.  A preview on how things end up looking for me on Windows with the change:

![keybindings_size](https://github.com/napari/napari/assets/16781833/93b95872-4941-49a3-911f-3a825f5b0b7f)

However, one thing that could be thought as a regression with the change above is that it will be impossible to resize the columns manually (so no mouse pointer change indicating a posible column resize when passing the mouse over the column header edge).

Following that caveat, not totally sure if the change above is a good enough fix, but anyhow, pushed a branch with the change in case you want to check/play with it: https://github.com/napari/napari/compare/main...dalthviz:napari:issue_6601

Let me know what do you think!

**Comment by Czaki** [comment]
It looks related to this bug in Qt: https://bugreports.qt.io/browse/QTBUG-120418

In the comments, there is information about setting parentship that solves the problem.

**Comment by dalthviz** [comment]
Gave the Qt bug link a check and although changing the parent of the `QTableWidget` didn't change the behavior (creating an intermediate widget to set as the table parent) seems like indeed something regarding size definition is breaking when you set programatically a stylesheet for widgets. Commenting out the `setStyleSheet` calls that are done to the table/table header instances at:

https://github.com/napari/napari/blob/374caa6ea37428d38b8240742fe142714c6fe763/napari/_qt/widgets/qt_keyboard_settings.py#L206-L208

https://github.com/napari/napari/blob/374caa6ea37428d38b8240742fe142714c6fe763/napari/_qt/widgets/qt_keyboard_settings.py#L240-L241

helps setting the correct columns size once you manually trigger the table regeneration (by selecting a different group of shortcuts):

![keybindings_size_style](https://github.com/napari/napari/assets/16781833/6b5c0139-f19b-489f-a997-09a8b2af5c68)

Quite a strange behavior so indeed seems like a Qt bug, right?

**Comment by psobolewskiPhD** [comment]
This is fixed with `PyQt6_Qt6-6.6.3` -- note PyQt6 is still 6.6.1
